qiskit/providers/ibmq/api/rest/job.py

```diff
@@ -63,6 +66,7 @@ def get(self) -> Dict[str, Any]:
         url = self.get_url('self')
 
         response = self.session.get(url).json()
+        logger.debug('Retrieved job %s. Endpoint: %s', self.job_id, url)
```

Comment:
These log messages are nice to have, but if you don't want to add logging to every method, you could just log in `session` (which should be done regardless) since a lot can be inferred from the url + request data.

Reply:
I think `session.request()` is a nice, centralized place for logging. It also allows you to log extra information easily.  If you think there would be too much noise, you can always choose not to log for some endpoints (such as `auth`).